devito/archinfo.py

```diff
@@ -107,15 +113,31 @@ def get_cpu_brand():
         # Fallback 1: it should now be fine to use psutil
         physical = psutil.cpu_count(logical=False)
         if not physical:
-            # Fallback 2: we might end up here on more exotic platforms such a Power8
-            # Hopefully we can rely on `lscpu`
             try:
-                physical = lscpu()['Core(s) per socket'] * lscpu()['Socket(s)']
+                lscpu_mapper = lscpu()
+                # Fallback 2: we might end up here
+                # on more exotic platforms such a Power8 or due to
+                # erroneous autodetection
+                physical = (lscpu_mapper['Core(s) per socket']
+                            * lscpu_mapper['Socket(s)'])
             except KeyError:
                 warning("Physical core count autodetection failed")
                 physical = 1
-    cpu_info['physical'] = physical
 
+    try:
+        if cpu_info['brand'] == 'aarch64':
```

Comment:
```
if cpu_info['brand'] in ('arm', 'aarch64'):
```
?

Reply:
What is the issue here? Older ARMs will return arm, while in cases we are not able to autodetect and get
the raw_arch_string we get aarch64. Let me know if we need more insight here